WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergGalleryUploadProcessor.swift

```diff
@@ -28,76 +28,67 @@ class GutenbergGalleryUploadProcessor: Processor {
+            let classAttributes = imgClass.components(separatedBy: " ")
 
-        guard let imageIDAttribute = classAttributes.filter({ (value) -> Bool in
-            value.hasPrefix(GutenbergImgUploadProcessor.imgClassIDPrefixAttribute)
-        }).first else {
-            return nil
-        }
+            guard let imageIDAttribute = classAttributes.filter({ (value) -> Bool in
+                value.hasPrefix(GutenbergImgUploadProcessor.imgClassIDPrefixAttribute)
+            }).first else {
+                return
+            }
 
-        let imageIDString = String(imageIDAttribute.dropFirst(ImageKeys.classIDPrefix.count))
-        let imgUploadID = Int32(imageIDString)
+            let imageIDString = String(imageIDAttribute.dropFirst(ImageKeys.classIDPrefix.count))
+            let imgUploadID = Int32(imageIDString)
 
-        guard imgUploadID == self.mediaUploadID else {
-            return nil
-        }
+            guard imgUploadID == self.mediaUploadID else {
+                return
+            }
 
-        let newImgClassAttributes = imgClass.replacingOccurrences(of: imageIDAttribute, with: ImageKeys.classIDPrefix + String(self.serverMediaID))
+            let newImgClassAttributes = imgClass.replacingOccurrences(of: imageIDAttribute, with: ImageKeys.classIDPrefix + String(self.serverMediaID))
 
-        var attributes = img.attributes
-        attributes.set(.string(self.remoteURLString), forKey: "src")
-        attributes.set(.string(newImgClassAttributes), forKey: "class")
-        attributes.set(.string("\(self.serverMediaID)"), forKey: ImageKeys.dataID)
-        attributes.set(.string(self.remoteURLString), forKey: ImageKeys.dataFullURL)
-        if attributes.contains(where: { $0.key == ImageKeys.dataLink } ) {
-            attributes.set(.string(self.mediaLink), forKey: ImageKeys.dataLink)
-        }
+            _ = try? imgTag.attr("src", self.remoteURLString)
+            _ = try? imgTag.attr("class", newImgClassAttributes)
+            _ = try? imgTag.attr(ImageKeys.dataID, "\(self.serverMediaID)")
+            _ = try? imgTag.attr(ImageKeys.dataFullURL, self.remoteURLString)
 
-        var html = "<\(ImageKeys.name) "
-        let attributeSerializer = ShortcodeAttributeSerializer()
-        html += attributeSerializer.serialize(attributes)
-        html += " />"
-        return html
-    })
+            let dataLinkAttribute = try? imgTag.attr(ImageKeys.dataLink)
+            if dataLinkAttribute != nil {
+                _ = try? imgTag.attr(ImageKeys.dataLink, self.mediaLink)
+            }
+        }
+    }
 
     private struct LinkKeys {
         static let name = "a"
     }
 
-    lazy var linkPostMediaUploadProcessor = HTMLProcessor(for: LinkKeys.name, replacer: { (link) in
-
-        guard let linkOriginalContent = link.content else {
-            return nil
-        }
+    func processLinkPostMediaUpload(_ block: GutenbergParsedBlock) {
+        let aTags = try? block.elements.select(LinkKeys.name)
```

Comment:
Same comment about adding an `if let` or `guarding` to remove the optional.

Reply:
I applied this suggestion in https://github.com/wordpress-mobile/WordPress-iOS/pull/22896/commits/d60c1d19412f8df4723a174ab3982ea0af4fe6f1.